### ISSUE TITLE
Improve the documentation for env vars: clarify why the env-var-name property shouldn't be used with {{ .Env.X }}

### ISSUE THREAD

**Issue by didiladi** [dynatrace-oss/dynatrace-monitoring-as-code#51]
See the following comment [here](https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/pull/49#issuecomment-730195774): 

```
Because the `env-token-name` property is implemented in a way to serve as a usability improvement. The actual token is read from the environment variable as soon as something is applied to an environment and not before that. This ensures, that:
* a dry-run doesn't need any tokens
* you only need to provide the token for the environment you want to apply the configs to

E.g.: in your `environments.yaml` you defined 2 environments (dev and prod) and you just want to apply the configs to dev. To do that, you don't need to have the token for prod set as an environment variable.
```

**Comment by CruzanCaramele** [comment]
@didiladi is this resolved from your PR ?

**Comment by didiladi** [comment]
> @didiladi is this resolved from your PR ?

No, we still need to document in the readme what I wrote in [this comment](https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/pull/49#issuecomment-730195774).

**Comment by robertjahn** [comment]
If you are in the README -- can you SORT this table by the first column?

![image](https://user-images.githubusercontent.com/24907089/102424108-7dbdab00-3fd8-11eb-8311-f419fae2b09b.png)

**Comment by patrickpichler** [comment]
Since monaco v2 no longer supports the `{{ .Env.X }}` syntax, this 
should be fixed by #351.

**Comment by CruzanCaramele** [comment]
will be fixed by https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/issues/351
